File: old_trunk/cost/iona/orbacus/cpp/main/TestObject.cpp
// **********************************************************************
//
// Copyright (c) 2001
// IONA Technologies, Inc.
// Waltham, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include "TestObject.h"

TestObject::TestObject(
    CORBA::ORB_ptr orb,
    ORBTest::Intf_ptr test_intf
)
    : m_orb(CORBA::ORB::_duplicate(orb)),
      m_test_intf(ORBTest::Intf::_duplicate(test_intf))
{
}

TestObject::~TestObject()
{
}

File: cost/iona/orbacus/java/main/TestObject.java
// **********************************************************************
//
// Copyright (c) 2001
// IONA Technologies, Inc.
// Waltham, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package ORBTest;

public interface TestObject
{
    boolean
    is_supported(
	org.omg.CORBA.Object obj
    );

    void
    test_SII(
	org.omg.CORBA.Object obj
    );

    void
    test_DII(
	org.omg.CORBA.Object obj
    );
}
